crates/observe/src/request_id.rs

```
@@ -95,43 +74,154 @@ macro_rules! make_service_with_task_local_storage {
     }};
 }
 
+/// Looks up the request id from the current tracing span.
+pub fn from_current_span() -> Option<String> {
+    let mut result = None;
+
+    Span::current().with_subscriber(|(id, sub)| {
+        let registry = sub.downcast_ref::<Registry>().unwrap();
```

Does `downcast_ref()` always succeed in this case?
If not we could return an error or change to `.expect()`.

---

Yes, this should never fail. Since it's easy enough I added a helpful error log instead of the unwrap().
I decided against the `.expect()` since the absence of the feature probably shouldn't abort the whole process.